src/wp-includes/option.php

```diff
@@ -2893,3 +2877,50 @@ function filter_default_option( $default_value, $option, $passed_default ) {
 
 	return $registered[ $option ]['default'];
 }
+
+/**
+ * Check if two database values are equal.
+ *
+ * @since 6.4.0
+ * @access private
+ *
+ * @param mixed $old_value The old value to compare.
+ * @param mixed $new_value The new value to compare.
+ * @return bool True if the values are equal, false otherwise.
+ */
+function _is_equal_database_value( $old_value, $new_value ) {
+	$values = array(
+		'old' => $old_value,
+		'new' => $new_value,
+	);
+
+	foreach ( $values as $_key => &$_value ) {
+		// Special handling for false-ish values.
+		if ( false === $_value ) {
+			$_value = '0';
+
+		// phpcs:ignore Generic.WhiteSpace.ScopeIndent.Incorrect
+		// Empty strings in the database should be seen as equivalent to false-ish cache values.
+		} elseif ( 'old' === $_key && '' === $_value ) {
+			$_value = '0';
+
+		// phpcs:ignore Generic.WhiteSpace.ScopeIndent.Incorrect
+		// Cast scalars to a string so type discrepancies don't result in cache misses.
+		} elseif ( is_scalar( $_value ) ) {
+			$_value = (string) $_value;
+		}
```

Comment:
Let's avoid the `phpcs:ignore` statements, the comments can be rearranged in a way that allows us to use correct indentation. I would suggest you put together the `if ... elseif ... elseif` without any comments in between, and instead put an overarching comments covering all three conditions above it (using a multiline comment `/* ...`).